drivers/hmis_external_apis/app/models/hmis_external_apis/mci_payload.rb

```diff
@@ -0,0 +1,144 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HmisExternalApis
+  class MciPayload
+    def self.from_client(client)
+      # FIXME: I need more documentation on what's required and how to
+      # correctly leave things out. Various attempts at partial payloads result
+      # in error messages that aren't helpful like:
+      #   "Object reference not set to an instance of an object."
+      #   "Length cannot be less than zero. (Parameter 'length')"
+      #
+      #   What do we send when we don't have a date? null doesn't seem to work nor the empty string.
+      #   Perhaps it's: "0001-01-01T00:00:00"
```

Comment:
`birthDate` is required for `ClientInsertVM` and `ClientUpdateVM` payload schemas. So if that is the date you're having issues with, that's why. We should error if any of `client.first_name` `client.last_name` or `client.dob` are missing.

Reply:
It was more a general problem as I reversed engineered the API. We don't get a "birthDate was invalid" error but instead useless errors that don't help know the problem.